components/base/kinematicbase/execution.go

```diff
@@ -466,6 +466,9 @@ func (ptgk *ptgBaseKinematics) courseCorrect(
 			connectionPoint.durationSeconds *= pctTrajRemaining
 			connectionPoint.subTraj = connectionPoint.subTraj[solution.trajIdx:]
 
+			// set the end position of the end position of the arcstep we were just on to the end position of the corrective trajectory
+			arcSteps[arcIdx].arcSegment.EndPosition = correctiveArcSteps[0].arcSegment.EndPosition
```

Comment:
`arcSteps[arcIdx].arcSegment.EndPosition = correctiveArcSteps[0].arcSegment.EndPosition`

This should be

`arcSteps[arcIdx].arcSegment.EndPosition = correctiveArcSteps[0].arcSegment.StartPosition`